internal/controller/pdbwatcher_controller.go

```diff
@@ -106,10 +127,14 @@ func (r *PDBWatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 
 	// Update status dynamically
-	pdbWatcher.Status.CurrentReplicas = *deployment.Spec.Replicas
+	if pdbWatcher.Status.CurrentReplicas == 0 {
```

Comment:
why the == 0 check and how is current replicas different from min? 

Doesn't have to be in this pr but you need a way to know that someone else besides this controller changed replicas